projects/graphql-client/src/graphql-request.service.test.ts

```diff
@@ -90,159 +154,143 @@ describe('GraphQl Request Service', () => {
-    spectator.expectOne(graphQlUri, HttpMethod.POST).flush(buildServerResponse());
-    // Make second identical request - should not be cached
-    spectator.service.queryImmediately(buildRequest()).subscribe();
-    spectator.expectOne(graphQlUri, HttpMethod.POST).flush(buildServerResponse());
-    expect(handler.getRequestOptions).toHaveBeenCalledTimes(2);
-  });
+    spectator.service.query(buildRequestOne()).subscribe();
+    const queryFnMock = mockQueryMethod();
+    tick(debounceTimeMs);
+    expect(queryFnMock).toHaveBeenCalledWith({
+      query: gql(
+        new GraphQlRequestBuilder()
+          .withSelects({ path: 'testone', children: [{ path: 'id' }, { path: 'value' }] })
+          .build()
+      ),
+      errorPolicy: 'all',
+      fetchPolicy: 'no-cache'
+    });
+    expect(handler.getRequestOptions).toHaveBeenCalledTimes(1);
+  }));
 
-  test('supports custom request options from mutation request handlers', () => {
-    const handler = buildMutationHandler({ cacheability: GraphQlRequestCacheability.NotCacheable });
+  test('supports custom request options from mutation request handlers', fakeAsync(() => {
+    const handler = buildMutationHandlerOne({ cacheability: GraphQlRequestCacheability.NotCacheable });
     spectator.service.registerHandler(handler);
-    spectator.service.mutate(buildRequest()).subscribe();
-    spectator.expectOne(graphQlUri, HttpMethod.POST).flush(buildServerResponse());
-    // Make second identical request - should not be cached
-    spectator.service.mutate(buildRequest()).subscribe();
-    spectator.expectOne(graphQlUri, HttpMethod.POST).flush(buildServerResponse());
-    expect(handler.getRequestOptions).toHaveBeenCalledTimes(2);
-  });
-
-  test('late subscribers to a debounced query can access results', fakeAsync(() => {
-    spectator.service.registerHandler(buildQueryHandler());
-    const query$ = spectator.service.queryDebounced(buildRequest());
-    tick(50); // Late subscriber
-    spectator.controller.expectNone(graphQlUri, HttpMethod.POST);
-    let result;
-    const subscription = query$.subscribe(res => (result = res));
-    spectator.controller.expectNone(graphQlUri, HttpMethod.POST);
+    spectator.service.mutate(buildRequestOne()).subscribe();
+    const mutateFnMock = mockMutateMethod();
     tick(debounceTimeMs);
-    spectator.controller.expectOne(graphQlUri, HttpMethod.POST).flush(buildServerResponse());
-    tick();
-    expect(result).toEqual(expect.objectContaining({ id: 'foo' }));
-    expect(subscription.closed).toBe(true);
+    expect(mutateFnMock).toHaveBeenCalledWith({
+      mutation: gql(
+        `mutation ${new GraphQlRequestBuilder()
+          .withSelects({ path: 'testone', children: [{ path: 'id' }, { path: 'value' }] })
+          .build()}`
+      )
+    });
+    expect(handler.getRequestOptions).toHaveBeenCalledTimes(1);
   }));
 
-  test('resubscribing to an immediate query triggers an immediate new request', () => {
```

Comment:
Where did the new request on new subscription tests go? These should still be verified with a mocked apollo.

Reply:
Pushing another commit with these tests. Almost done